src/modules/vtol_att_control/vtol_att_control_main.cpp

```diff
@@ -461,7 +464,20 @@ VtolAttitudeControl::Run()
 		_local_pos_sub.update(&_local_pos);
 		_local_pos_sp_sub.update(&_local_pos_sp);
 		_pos_sp_triplet_sub.update(&_pos_sp_triplet);
-		_airspeed_validated_sub.update(&_airspeed_validated);
+		bool got_airspeed = _airspeed_validated_sub.update(&_airspeed_validated);
+		if (PX4_ISFINITE(_calibrated_airspeed_m_s_smoothed) && _params.vt_aspd_tau > 0.1f) {
+			// Smooth the airspeed value using exponential moving average
+			float alpha = 1 - expf(-dt_us / (_params.vt_aspd_tau * 1e6f));
+			_calibrated_airspeed_m_s_smoothed = _calibrated_airspeed_m_s_smoothed * (1-alpha) + _airspeed_validated.true_airspeed_m_s * alpha;
```

Comment:
Doesn't PX4 have existing functionality that you can use for low pass filtering? There are some gotchas (initialization, stale values, etc) which it would be nice to not re-invent

Reply:
I recommend AlphaFilter from the maths library

Reply:
Also: Future maintaining Herman wonders where the value 0.1 comes from and if it can be changed

Reply:
Yes, should choose a standard implementation, which allows values all the way down to zero safely.